### ISSUE TITLE
Convert grayscale layer or channel to transparency

### ISSUE THREAD

**Issue by Woovie** [photopea/photopea#6329]
I'm working with a 3rd party solution in a game engine that requires me to compose what would normally be an RGB and grayscale image into a single RGBA image with the grayscale portion put in the alpha channel for exporting to DDS or PNG. I'd like a feature in Photopea to right click a layer -> convert to alpha channel or some solution for this.

I may have missed if this is possible otherwise. I've tried a few things to no avail. Primarily, I created the "Alpha 1" channel, pasted the grayscale image, which did work. But, exporting to PNG discarded this channel. I'm also open to the idea of converting a channel to the transparency channel. I am open to what would be best for the Photopea team to enable some way to take a grayscale image and make it the transparency.

Thank you for your time.

**Comment by photopea** [comment]
When you export to DDS, and your image has no transparency, and there is an extra channel in "Channels", that channel will be attached as the four channel inside the DDS.

**Comment by Woovie** [comment]
Thanks for that, I am primarily working with PNGs as DDS is sadly not very compatible across various platforms which a game can be distributed.

In the case of PNG, I don't have any solution. I recently found a solution in Photoshop that perhaps could be implemented here for PNG users. I'd like to load the selection from the gray of a 16bit image. Photoshop does it like so:

![image](https://github.com/photopea/photopea/assets/7304619/702e1d28-fd45-4f74-9dcc-0233d67f0905)

This would provide users who I've seen ask similar questions as well, while also providing feature parity.

**Comment by photopea** [comment]
We have the same feature in Photopea, it works the same as in Photoshop.

**Comment by Woovie** [comment]
> We have the same feature in Photopea, it works the same as in Photoshop.

Is it limited in some way? I could not get it to work. I can only select background transparency. I cannot select the gray channel.

![image](https://github.com/photopea/photopea/assets/7304619/696dc0a6-8a06-4189-8244-7a4aeebc28fa)

**Comment by photopea** [comment]
It will work the same in Adobe Photoshop. You need to create additional channels (in Photoshop or in Photopea). Maybe look into some book about Adobe Photoshop, or ask someone here: https://www.reddit.com/r/photopea/ to write you a tutorial. This is not a place for teaching users how to use Photopea :(

**Comment by Woovie** [comment]
It absolutely does not work like Photoshop. I posted a Photoshop screenshot above clearly showing it working, and Photopea not replicating that behavior. I understand the impetus is on the user to understand how to use the tool, but this is not feature parity. If I find a solution I'll gladly post it here for future users who are also facing the issue of the feature not being parity.

**Comment by photopea** [comment]
Could you be more precise? What did you do to see what you saw in Photoshop? Did you try to do the same steps in Photopea?

**Comment by Woovie** [comment]
Select -> Load Selection -> Choose black and white image loaded, the "Channel" option is "Gray" for Photoshop. This is using the 16bit grayscale image as a transparency selector. Photoshop is detecting that it is a grayscale image specifically and providing the functionality to use that as a transparency select.

**Comment by photopea** [comment]
I opened Photoshop, all options inside "Select" are inactive.

When I open a JPG photo of myself in Photoshop, Select - Load Selection is inactive. 

Could you be more precise?

**Comment by Woovie** [comment]
> I opened Photoshop, all options inside "Select" are inactive, because no file is opened.

Pedantry is unnecessary, I've paid for a license multiple times in the past and respect the tools you've built. I've used Photoshop for years and you've built a tool to replicate its fundamental features, so you are well aware that the intent is to have an image open already.

> When I open a JPG photo of myself in Photoshop, Select - Load Selection is inactive.

My initial reply indicates very clearly I'm combining two images:

> what would normally be an RGB and grayscale image into a single RGBA image with the grayscale portion put in the alpha channel

The fundamental issue is that Photopea treats grayscale images as a 3 channel RGB image instead of a single channel grayscale image. Reading the documentation of UPNG.js, it is claimed to support grayscale:

https://github.com/photopea/UPNG.js/blob/master/README.md?plain=1#L45

You can review my prior replies providing an example image pack download link: https://ambientcg.com/view?id=Ground071

As well, my prior provided screenshots show which images I have open, the color and displacement images. Color is RGB PNG 8bit and displacement is 16bit grayscale.

To replicate my issue, download the 1k PNG variants. The exact filenames are:

- `Ground071_1K-PNG_Color.png`
- `Ground071_1K-PNG_Displacement.png`

Photoshop detects `Ground071_1K-PNG_Displacement.png` correctly as grayscale and provides a single channel, "Gray", as expected:

![image](https://github.com/photopea/photopea/assets/7304619/dff48b7d-80ad-44c5-a5db-d3bda0bd62a9)

Photopea loads this image as an RGB image:

![image](https://github.com/photopea/photopea/assets/7304619/bdefbc9b-7ef2-4f4d-9e9f-bdc0165a5562)

If Photopea truly supports grayscale images as it library claims to, it should have a grayscale mode as Photoshop does:

![image](https://github.com/photopea/photopea/assets/7304619/c16561c6-6339-4e3a-9401-e65e0d7488d2)

![image](https://github.com/photopea/photopea/assets/7304619/4e4edde7-1a6c-45d3-bf55-1e3c2b60988e)

If grayscale is properly supported, this issue I filed would be relevant, but I didn't dig into the deeper parts of Photopea prior as it seemed like an unsupported feature, but it was one step deeper being lack of full grayscale support.

I fully understand support is hard, software is complex, and users don't read. Showing some empathy and being patient with a user is hard, I know that best having worked support for many years in my past at a webhost doing everything from Wordpress basics to root user system configuration. Please understand most users do not have ill intent and are simply looking to see if Photopea could add a feature to improve a workflow.

**Comment by photopea** [comment]
Yes, I understand, I have opened both of your images in Adobe Photosohp.

But what do you do next? Select - Load Selection is still not available.

I am just trying to figure out, if what you need is possible in Adobe Photoshop or in any other software.

**Comment by MichalSZZ** [comment]
I think the main of Photopea is not having a Grayscale color mode. I wrote about that in other issues about printing on real print machines (not home printers). When I have print samples the page was made in Indesign in grayscale (real grayscale) and the printer used then only K (black) when the images were printed. And the second one page (the same) was in CMYK mode with a black and white photo, and the printer used all CMYK colors to simulate B&W.

Here we have a problem with something specific only for the game industry. But look for your post on FB Ivan - I think it is a big part of the cake in the global Photopea user percentage. 

I tried now the whole process in Photoshop and id doesn't look like in Photopea. I opened two files (color and displacement).
On the first look I see that Displacement PNG file is in Grayscale color mode with only one Channel - named Grayscale.
![obraz](https://github.com/photopea/photopea/assets/38118273/e9f7ce16-ac94-4937-ad13-22964fd918d9)

When I open Load selection on the first tab (color RGB image) and I am trying to choose the second document, I see a Grayscale channel to load as a selection.

![obraz](https://github.com/photopea/photopea/assets/38118273/caa5bfb2-dde1-44f3-b020-69fb40243a11)

Then I tried to apply this selection as a mask to the main layer. And the effect is:
![obraz](https://github.com/photopea/photopea/assets/38118273/08254f62-e363-482f-8d74-cb03d9801406)

In Photopea: 

- there isn't one channel for this grayscale displacement file, it loads this file as a RGB image with RGB channels
- if I try to load selection from Displacement grayscale file I have one option - 'background transparency' (OK maybe it is flattened one channel that should be named Grayscale - I don't know
- it load the selection, but when I tried to add mask in the same way - the effect was a little bit different (I don't know why but it Was)

But the funnies thing is... now when I write this - I doesn't work, like 20 minutes ago. It doesn't work at all. I can't load channel to color image anymore.
